charts/maintenance-page/templates/_helpers.tpl

```diff
@@ -0,0 +1,66 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "maintenance-page.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "maintenance-page.fullname" -}}
+{{- if .Values.environment }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- printf "%s-%s" .Values.environment $name | trunc 63 | trimSuffix "-" }}
+{{- else if .Values.fullnameOverride }}
```

Comment:
Maybe you should check for `fullnameOverride` first? Because now if there is a `environment` param in values, we just ignore that. But the case when we need to use `fullnameOverride` doesn't mean that we won't have an `environment`. 

So I think we should first check whether the maintenance page has `fullnameOverride` and use it if it exists, then we should check whether `environment` exists and create the name from environment and move further

Reply:
This makes total sense. Done - pls review.